src/main/java/org/synyx/urlaubsverwaltung/absence/OvertimeAbsenceApiController.java

```diff
@@ -0,0 +1,86 @@
+
+import static java.lang.invoke.MethodHandles.lookup;
+import static org.slf4j.LoggerFactory.getLogger;
+import static org.springframework.hateoas.MediaTypes.HAL_JSON_VALUE;
+import static org.springframework.http.HttpStatus.*;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.synyx.urlaubsverwaltung.absence.AbsenceApiController.ABSENCES;
+import static org.synyx.urlaubsverwaltung.security.SecurityRules.IS_BOSS_OR_OFFICE;
+
+@Tag(
+    name = "absences"
+)
+@RestControllerAdviceMarker
+@RestController
+@RequestMapping("/api/persons/{personId}")
+public class OvertimeAbsenceApiController {
+
+    private static final Logger LOG = getLogger(lookup().lookupClass());
+
+    private final ApplicationService applicationService;
+
+    @Autowired
+    public OvertimeAbsenceApiController(ApplicationService applicationService) {
+        this.applicationService = applicationService;
+    }
+
+    @Operation(
+        summary = "Returns the amount of overtime used for a specific absence of category OVERTIME",
+        description = """
+            Returns the amount of overtime used for a specific absence of category OVERTIME.
+
+            Needed basic authorities:
+            * user
+
+            Needed additional authorities:
+            * user                   - if the requested absences of the person id is from the authenticated user
+            * department_head        - if the requested absences of the person id is a managed person of the department head and not of the authenticated user
+            * second_stage_authority - if the requested absences of the person id is a managed person of the second stage authority and not of the authenticated user
+            * boss or office         - if the requested absences of the person id is any id but not of the authenticated user
+            """
+    )
+    @GetMapping(value = ABSENCES + "/{absenceId}/overtime", produces = {APPLICATION_JSON_VALUE, HAL_JSON_VALUE})
+    @PreAuthorize(IS_BOSS_OR_OFFICE +
+        " or @userApiMethodSecurity.isSamePersonId(authentication, #personId)" +
+        " or @userApiMethodSecurity.isInDepartmentOfDepartmentHead(authentication, #personId)" +
+        " or @userApiMethodSecurity.isInDepartmentOfSecondStageAuthority(authentication, #personId)")
+    public OvertimeAbsenceDto overtimeAbsence(
+        @Parameter(description = "ID of the person")
+        @PathVariable("personId")
+        Long personId,
+        @Parameter(description = "ID of the absence")
+        @PathVariable("absenceId")
+        Long absenceId) {
+        ResponseStatusException notFoundException = new ResponseStatusException(NOT_FOUND, "No absence for personId=" + personId + " and absenceId=" + absenceId);
+        Application application = applicationService.getApplicationById(absenceId).orElseThrow(() -> notFoundException);
+
+        if (!application.getPerson().getId().equals(personId)) {
+            throw notFoundException;
+        }
```

Comment:
fragen die ich mir beim lesen gestellt habe:
- warum ist die exception in einer variablen abgelegt?
- warum wird `application.getPerson` noch mal gecheckt? es gibt doch ein `PreAuthorize` (ja, dumme frage... es geht darum ob die application auch zur person passt, aber der Gedanke kam mir direkt...)

dinge auslagern zu wollen in Methoden hilft beim schreiben von (imho) leserlichen code. was wir hier wollen ist die `Application` zu einer `person` und `applicationId` zu bekommen.

also entweder eine extra methode dafür, oder im kopf anfangen mit `return` und mit einer fluent api arbeiten:

```java
applicationOfPerson = applicationService.getApplicationById(absenceId)
    .filter(application -> application.getPerson().getId().equals(personId))
    .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "message"));
```

Reply:
Ja, guter Vorschlag. Passe ich so an.

Meine initialen Gedanken waren:
- PreAuthorize checked erstmal nur, ob die Request gemacht werden darf, aber garnicht, ob die angefragte Application zur Person gehört. Dann könnte $Angreifer mit `personId=AngreiferId` und `absenceId=fremdeAbsenceId` Infos zur Absence einer anderen Person abfragen. 
- Wenn die Fehlermeldungen in den zwei Fällen verschieden sind, kann $Angreifer rausfinden, ob es eine Application mit der Id überhaupt gibt. (Unsicher, ob das in de UV als schützenswert gilt)